src/components/LivestreamSection/index.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  Image,
+  TouchableOpacity,
+  View,
+  Text,
+  ImageSourcePropType,
+} from 'react-native';
+import React, { useEffect, useState, useCallback } from 'react';
+import { FileRepository, StreamRepository } from '@amityco/ts-sdk-react-native';
+import { useStyles } from './styles';
+import { useNavigation } from '@react-navigation/native';
+import { RootStackParamList } from '../../routes/RouteParamList';
+import { SvgXml } from 'react-native-svg';
+import { exclamationIcon, playBtn } from '../../svg/svg-xml-list';
+import LivestreamEndedView from './LivestreamEndedView';
+import type { NativeStackNavigationProp } from '@react-navigation/native-stack';
+
+interface ILivestreamSection {
+  streamId: Amity.Stream['streamId'];
+}
+
+const LivestreamSection: React.FC<ILivestreamSection> = ({ streamId }) => {
+  const styles = useStyles();
+
+  const navigation =
+    useNavigation<
+      NativeStackNavigationProp<RootStackParamList, 'VideoPlayer'>
+    >();
+
+  const [livestream, setLivestream] = useState<Amity.Stream>();
+  const [thumbnailUrl, setThumbnailUrl] = useState<ImageSourcePropType>();
+
+  const onPlayLivestream = useCallback(() => {
+    if (!livestream) return;
+
+    navigation.navigate('LivestreamPlayer', { streamId: livestream.streamId });
+  }, [livestream, navigation]);
+
+  const getLivestreamThumbnail = async (currentStream: Amity.Stream) => {
+    const defaultThumbnail = require('../../../assets/images/default-livestream-thumbnail.png');
+
+    if (currentStream.thumbnailFileId) {
+      const file = await FileRepository.getFile(currentStream.thumbnailFileId);
+
+      if (file) {
+        const fileUrl = FileRepository.fileUrlWithSize(
+          file.data.fileUrl,
+          'full'
+        );
+
+        setThumbnailUrl({ uri: fileUrl });
+        return;
+      }
+    }
+    setThumbnailUrl(defaultThumbnail);
+  };
+
+  useEffect(() => {
+    const getLivestream = () => {
+      return StreamRepository.getStreamById(
+        streamId,
+        ({ data, loading, error }) => {
+          if (error) console.error('Error fetching livestream', error);
+          if (!loading && data) {
+            setLivestream({ ...data });
+            getLivestreamThumbnail(data);
+          }
+        }
+      );
+    };
+
+    const unsubscribe = getLivestream();
```

Comment:
don't need to assign getLivestream() to unsubscribe. 
Assigning function with invoke syntax`()` to a variable calls  immediately. 
It means we are unsubscribe immediately. Not only in useEffect clean up. 

just use getLivestream() in useEffect return directly.